### PR TITLE
Upgrade to Java 17 (DTSBPS-1071)

### PR DESCRIPTION
**Pull request by alivenichoppa**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSBPS-1071

### Change description ###
Upgrade to Java 17

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by apereverzin — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ RUN wget -P /tmp https://github.com/microsoft/ApplicationInsights-Java/releases/
 
 # Application image
 
-FROM hmctspublic.azurecr.io/base/java:openjdk-11-distroless-1.4
+FROM hmctspublic.azurecr.io/base/java:17-distroless
 
 COPY --from=downloader /tmp/applicationinsights-agent-${APP_INSIGHTS_AGENT_VERSION}.jar /opt/app/
```
Comment (mdayican) [inline Dockerfile:13]: need custom applicationinsights.json
no need to copy jar.

Comment (mdayican) [reply Dockerfile:13]: you can delete lib/ai.xml

Comment (mdayican) [reply Dockerfile:13]: ARG APP_INSIGHTS_AGENT_VERSION should be 3.X
sth :
ARG APP_INSIGHTS_AGENT_VERSION=3.2.6

File: lib/applicationinsights.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "connectionString": "${file:/mnt/secrets/reform-scan/app-insights-connection-string}",
```
Comment (mdayican) [inline lib/applicationinsights.json:2]: this log to bulk-scan ?
first in bulk-scan-shared infra you need to create "app-insights-connection-string" secret.

Comment (alivenichoppa) [reply lib/applicationinsights.json]: yes, it was created and merged https://github.com/hmcts/bulk-scan-shared-infrastructure/pull/272

**Review by mdayican — APPROVED** [review]

**Pull request merged**